Event Sourcing with Free Monad

speedcom opened the pull request.

speedcom committed:
using free-monad, first draft

speedcom committed:
moved out Blogger class to Domain object

rabbitonweb commented:
Will check it out on weekend :)

lukasz-golebiewski commented:
@speedcom Do you have any tests that would prove that this implementation actually works? ;-)

speedcom committed:
add Test Interpreter that return list of names of Events in proper order

speedcom commented:
@rabbitonweb @lukasz-golebiewski I've added additional tests to my proposal solution. To do that I've derivied dedicated Interpreter for test. Another one just check result of processing events by Pure Interpreter. Also cleaned a bit code.
